mlflow/gateway/providers/huggingface.py

```diff
@@ -0,0 +1,125 @@
+from typing import Any, Dict
+
+from fastapi import HTTPException
+from fastapi.encoders import jsonable_encoder
+
+from mlflow.gateway.config import HuggingFaceTextGenerationInferenceConfig, RouteConfig
+from mlflow.gateway.providers.base import BaseProvider
+from mlflow.gateway.providers.utils import (
+    rename_payload_keys,
+    send_request,
+)
+from mlflow.gateway.schemas import chat, completions, embeddings
+
+
+class HFTextGenerationInferenceServerProvider(BaseProvider):
+    def __init__(self, config: RouteConfig) -> None:
+        super().__init__(config)
+        if config.model.config is None or not isinstance(
+            config.model.config, HuggingFaceTextGenerationInferenceConfig
+        ):
+            raise TypeError(f"Unexpected config type {config.model.config}")
+        self.huggingface_config: HuggingFaceTextGenerationInferenceConfig = config.model.config
+        self.headers = {"Content-Type": "application/json"}
+
+    async def _request(self, path: str, payload: Dict[str, Any]) -> Dict[str, Any]:
+        return await send_request(
+            headers=self.headers,
+            base_url=self.huggingface_config.hf_server_url,
+            path=path,
+            payload=payload,
+        )
+
+    async def chat(self, payload: chat.RequestPayload) -> chat.ResponsePayload:
+        raise HTTPException(
+            status_code=404,
+            detail="The chat route is not available for the Text Generation Inference provider.",
+        )
+
+    async def completions(self, payload: completions.RequestPayload) -> completions.ResponsePayload:
+        payload = jsonable_encoder(payload, exclude_none=True)
+        self.check_for_model_field(payload)
+        key_mapping = {
+            "max_tokens": "max_new_tokens",
+        }
+        for k1, k2 in key_mapping.items():
+            if k2 in payload:
+                raise HTTPException(
+                    status_code=422, detail=f"Invalid parameter {k2}. Use {k1} instead."
+                )
+
+        # HF TGI does not support generating multiple candidates.
+        candidate_count = payload.get("candidate_count", 1)
+        if candidate_count != 1:
+            raise HTTPException(
+                status_code=422,
+                detail="'candidate_count' must be '1' for the Text Generation Inference provider."
+                f"Received value: '{candidate_count}'.",
+            )
+        prompt = payload.pop("prompt")
+        parameters = rename_payload_keys(payload, key_mapping)
+
+        # HF TGI does not support 0 temperature
+        parameters["temperature"] = max(payload["temperature"], 1e-3)
+        parameters["details"] = True
+        parameters["decoder_input_details"] = True
```

Comment:
What happens if a user supplies these parameters as `False` in their request payload? 

Can we attempt to get the values from the payload and set a default to `True` if not set by the user if these are required fields for TGI?

Reply:
Can this be added to the docs as a note explaining that these parameters are not available for configuration?